exatomic/gaussian/output.py

```diff
@@ -608,6 +619,212 @@ def parse_momatrix(self):
         if bcoefs is not None:
             self.momatrix['coef1'] = bcoefs
 
+    def parse_frequency_ext(self):
+        '''
+        Parses extended frequency data present in FChk file
+
+        Note:
+            Requires Freq(SaveNormalModes) in input
+        '''
+        # Frequency regex
+        _renmode = 'Number of Normal Modes'
+        _redisp = 'Vib-Modes'
+        _refinfo = 'Vib-E2'
+        _remass = 'Vib-AtMass'
+        # TODO: find out what these two values are useful for
+        _rendim = 'Vib-NDim'
+        _rendim0 = 'Vib-NDim0'
+
+        found = self.find(_renmode)
+        if not found:
+            return
+        else:
+            found = self.find(_renmode, _redisp, _refinfo, _remass, keys_only=True)
+        # find number of vibrational modes
+        nmode = self._intme(found[_renmode])
+        # get extended data (given by mapper array)
+        all_info = self._dfme(found[_refinfo], nmode * 14)
+        all_info[abs(all_info) < self.tol] = 0
+        # mapper array to filter through all of the values printed on the fchk file
+        # TODO: have to find labels of unknown columns
+        mapper = ["freq", "r_mass", "f_const", "ir_int", "unk1", "unk2", "unk3",
+                  "dipole_s", "rot_s", "unk4", "unk5", "unk6", "unk7", "em_angle"]
+        # build extended frequency table
+        self.frequency_ext = pd.DataFrame.from_dict({mapper[int(i/nmode)]: all_info[i:i+nmode]
+                                                     for i in range(0, nmode*14-1, nmode)})
+
+    def parse_frequency(self):
+        '''
+        Parses frequency data from FChk file
+
+        Note:
+            Requires Freq(SaveNormalModes) in input
+        '''
+        # Frequency regex
+        _renmode = 'Number of Normal Modes'
+        _redisp = 'Vib-Modes'
+
+        found = self.find(_renmode)
+        if not found:
+            return
+        else:
+            found = self.find(_renmode, _redisp, keys_only=True)
+        # get atomic numbers
+        znums = self.atom['Zeff'].values
+        # get number of atoms
+        nat = len(znums)
+        # TODO: will the order of the atoms and their frequencies always be the same?
+        # get frequency modes from extended data table
+        freq = self.frequency_ext['freq']
```

Comment:
Since the above parse function has the possibility of returning without setting this attribute, this has the potential to throw an `AttributeError`. I think this should be handled somehow.

Reply:
I guess technically it could, but like I said before it should not execute that far as it will not find any of the needed frequency flags. I was trying to save a bit on IO by doing this, but it may be more stable to just find it separately. The only reason that I split up the functions was that I could not get them to work so that two class variables can be set within one function. Is this something that we could change? Because I know with the parse_gradients we could merge into the parse_atom function and save on some coding.

Reply:
I think this comment deserves to out raised as an issue of the exa project. A minimal example would be good. I think you may have just highlighted a fundamental bug/limitation of the current 'typing' model of exa (Containers). Do you mind starting that issue up and I'll investigate more closely?

Reply:
I guess my question is more why we can't set two class attributes within one function. Looking through the code every function sets one and I was trying to set self.frequency and self.frequency_ext within the parse_frequency function. Don't know if that makes more sense.

Reply:
It does look like an issue.  There is an example of a workaround in the same file:
https://github.com/exa-analytics/exatomic/blob/6e4f62d43096a2fcf6d490489a2f8990808b133b/exatomic/gaussian/output.py#L267 
where `self.basis_set_order` is set inside of `parse_momatrix` and a dummy method `parse_basis_set_order` checks if it has been set and falls back to calling `parse_momatrix`.  Setting both attributes inside `parse_frequency` should in principle work.  I wonder if you are using the `to_universe` and the attribute is getting passed over when transferring attributes from the output to the uni.  This could just be a bug in that function, unless  of course, you aren't calling to_universe.

Reply:
I think I still don't understand exactly the issue. Isn't the quoted code there an example of a method that sets two class attributes? Here is a minimal working example of setting two attributes in a single method. Note the change in type (due to the `TypedMeta` metaclass machinery.
```
from exa import Editor, TypedMeta
import six

class Meta(TypedMeta):
    atom = int
    frame = float
    
class Fchk(six.with_metaclass(Meta, Editor)):
    def parse_atom(self):
        self.atom = 1.0
        self.frame = 1
        
f = Fchk("", ignore=True)
f.atom, type(f.atom), f.frame, type(f.frame)    # (1, int, 1.0, float)
```

Does that help explain the behavior you are seeing?